### PR TITLE
sql: deflake guardrails logictest by disabling distributed configs

### LINKED ISSUE #106526: pkg/sql/logictest/tests/fakedist/fakedist_test: TestLogic_guardrails failed
pkg/sql/logictest/tests/fakedist/fakedist_test.TestLogic_guardrails [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10851685?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10851685?buildTab=artifacts#/) on release-23.1 @ [a7d63b0fcba4a8874b171ee58c1c951575659057](https://github.com/cockroachdb/cockroach/commits/a7d63b0fcba4a8874b171ee58c1c951575659057):

```
=== RUN   TestLogic_guardrails
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/35399f9f3f2a119316a70107be975c37/logTestLogic_guardrails252806416
    test_log_scope.go:79: use -show-logs to present logs inline
[15:23:42] setting distsql_workmem='36826B';
    logic.go:2880: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12446/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails:34: SELECT * FROM guardrails
        expected:
        txn has read 51 rows, which is above the limit
        got:
        pq: txn has read 62 rows, which is above the limit: TxnID d1e322e5-f56a-4fcc-9e54-1397950f5d82 SessionID 17708aefb36041b30000000000000001
[15:23:42] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12446/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails with config fakedist: 9 tests, 1 failures
    logic.go:3992: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12446/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails:37: error while processing
    logic.go:3992: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12446/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails:37: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/35399f9f3f2a119316a70107be975c37/logTestLogic_guardrails252806416
--- FAIL: TestLogic_guardrails (1.59s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_guardrails.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-29605

### LINKED ISSUE #106787: pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test: TestLogic_guardrails failed
pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_guardrails [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10903701?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10903701?buildTab=artifacts#/) on master @ [b6b5d2812aad70c3f11649dc4292d6c2bdde2d77](https://github.com/cockroachdb/cockroach/commits/b6b5d2812aad70c3f11649dc4292d6c2bdde2d77):

```
=== RUN   TestLogic_guardrails
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/67345207899012758c6af8857533da11/logTestLogic_guardrails1005589471
    test_log_scope.go:81: use -show-logs to present logs inline
[20:33:02] setting distsql_workmem='11503B';
    logic.go:2907: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/13671/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails:34: SELECT * FROM guardrails
        expected:
        txn has read 51 rows, which is above the limit
        got:
        pq: txn has read 56 rows, which is above the limit: TxnID 3b736468-01bd-48fe-b3e2-ee48ae42107a SessionID 1771878ebef9b0370000000000000001
[20:33:02] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/13671/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails with config fakedist-disk: 9 tests, 1 failures
    logic.go:4047: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/13671/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails:37: error while processing
    logic.go:4047: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/13671/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails:37: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67345207899012758c6af8857533da11/logTestLogic_guardrails1005589471
--- FAIL: TestLogic_guardrails (1.65s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_guardrails.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-29715

### PR DESCRIPTION
**Pull request by rytaft**
Fixes #106526
Fixes #106787

Release note: None

### CODE REVIEW

**Review by yuzefovich — APPROVED** [review]
:lgtm:

> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/106791#-:-N_GNKWA2eli5sZ5Vzs0:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rytaft)

**Review by rytaft — COMMENTED** [review]
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/106791#-:-N_H48Hh15Oov1o-D8zG:b-oduykh)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rytaft)

**Comment by yuzefovich** [comment]
blathers backport 23.1
